controller/control.go

```diff
@@ -672,16 +678,19 @@ func (c *Controller) addReplicaDuringStartNoLock(address string) error {
 	)
 	newBackend, err := c.factory.Create(address)
 	if err != nil {
+		c.rmReplicaFromRegisteredReplicas(address)
```

Comment:
Can this be modelled as a `defer` logic.
This should check if err is not nil & do the de-registration

Reply:
@AmitKumarDas handling is different for the next set of errors in code below, which calls different methods.